extensions/ql-vscode/src/databases/config/db-config-store.ts

```diff
@@ -145,10 +145,38 @@ export class DbConfigStore extends DisposableObject {
     await this.writeConfig(config);
   }
 
+  public async addRemoteReposToList(
+    repoNwoList: string[],
+    parentList: string,
+  ): Promise<string[]> {
+    if (!this.config) {
+      throw Error("Cannot add variant analysis repos if config is not loaded");
+    }
+
+    const config = cloneDbConfig(this.config);
+    const parent = config.databases.variantAnalysis.repositoryLists.find(
+      (list) => list.name === parentList,
+    );
+    if (!parent) {
+      throw Error(`Cannot find parent list '${parentList}'`);
+    }
+
+    // Remove duplicates from the list of repositories.
+    const newRepositoriesList = [
+      ...new Set([...new Set(parent.repositories), ...new Set(repoNwoList)]),
+    ];
+
+    parent.repositories = newRepositoriesList.slice(0, 1000);
+    const truncatedRepositories = newRepositoriesList.slice(1000);
+
+    await this.writeConfig(config);
+    return truncatedRepositories;
```

Comment:
It would be more difficult for the caller to find out which items weren't added, because the list can have repositories from before. However, I could imagine returning a nested array with the first entry returning the added, the second the excluded items?

Reply:
I think that would be less readable, personally, so we can go with what you've got.